devex/cross-nuget/README.md

```diff
@@ -0,0 +1,136 @@
+
+1. Run `setup-host.sh`, accepting the LXD initialization defaults.
+2. Log out and log back in.
+3. Run `setup-containers.sh` to create and start the containers used for
+   building; unlike Docker containers, these remain running in the background
+   and have persistent storage; think of them as virtual machines instead of
+   containers.
+4. Clone the SDK:
+   ```bash
+   git clone -b v0.11.0 --recursive --depth=1 https://github.com/openenclave/openenclave sdk
+   ```
+5. Open `scripts/ansible/oe-contributors-setup.yml` and comment out the step
+   that installs the Intel SGX driver because drivers cannot be installed inside
+   containers.
+6. Install the prerequisites in each container:
+   ```bash
+   # Enter the Xenial container.
+   lxc exec oepkgxenial su ubuntu
+
+   cd /host/path/to/sdk
+   sudo scripts/ansible/install-ansible.sh
+   ansible-playbook scripts/ansible/oe-contributors-setup-cross-arm.yml
+
+   sudo apt install python3-pip p7zip-full -y
+   pip3 install pycryptodome
+
+   # Quit the Xenial container.
+   exit
+
+   # Enter the Bionic container.
+   lxc exec oepkgbionic su ubuntu
+
+   cd /host/path/to/sdk
+   sudo scripts/ansible/install-ansible.sh
+   ansible-playbook scripts/ansible/oe-contributors-setup-cross-arm.yml
+   sudo apt install python3-pyelftools p7zip-full -y
+
+   # Quit the Bionic container.
+   exit
+   ```
+
+## Building
+
+### Windows
+
+To build the SDK in all its configurations:
+
+1. Launch the x64 Native Tools Command Prompt for VS 2017 from the Start menu.
+2. Run:
+   ```cmd
+   cd %SYSTEMDRIVE%\path\to\this\repository\windows
+   powershell -c "Set-ExecutionPolicy Bypass -Scope Process; .\build.ps1"
+   powershell -c "Set-ExecutionPolicy Bypass -Scope Process; .\pack.ps1"
+   ```
+
+This creates a `Pack` directory with the build outputs.
+
+### Ubuntu
+To build the SDK in all its configurations, run:
+
+```bash
+cd /path/to/this/repository/linux
+./driver.sh
+./pack.sh
+```
+
+This creates a `pack` directory with the build outputs.
+
+## Packaging
+
+Fetch the resulting `pack` directories from both environments and merge their
+contents. This can be achieved by copying the contents of one folder into the
+other; there are no files that overlap.
+
+Then, copy the contents of the `devex/vsextension/msbuild/` directory into the
+`pack/build/native` directory. Additionally, copy the contents of the
+`devex/cross-nuget/extras` directory into the `pack` directory. Update the
+version information insde `open-enclave-cross.nuspec` if necessary.
+
+Lastly, create a `licenses` directory inside the merged `pack` directory and
+copy into it the following two files from the SDK:
+
+1. `LICENSE`
+2. `THIRD_PARTY_NOTICES`
+
+To create the NuGet package, navigate to the root of the merged `pack` directory
+and run:
+
+```bash
+# The extension informs 7z as to the type of archive to create.
+7z a -mm=Deflate -mx=9 -mmt=8 -mfb=258 -mpass=15 -r out.zip .
+mv out.zip open-enclave-cross.0.11.0-rc1-cbe4dedc-1.nupkg
```

Comment:
0.11.0-rc1-cbe4dedc-1.nupkg
How is cbe4dedc chosen? Is it always the same?
Is 0..11.0.rc1 for v0.11, RC1?

Reply:
Good catch! There was a time when I built a NuGet package for Dave off the tip of `master` instead of a tag, so to indicate which commit the package contained, I included the short commit hash in the name of the package. This is definitely not needed, so I've removed it.